problem1Spec.js

```diff
@@ -0,0 +1,41 @@
+describe("Problem1 test 1", function() {
+	var objectProblem1;
+	var testArray;
+	
+	beforeEach(function() {
+		objectProblem1 = new ProblemsSolved();
+	});
+  
+  it("Sum of even numbers in case of exist even number/numbers", function() {
+	testArray = [4,4,4,4,5];
+	expect(objectProblem1.SumOfEvenNumberInArray(testArray)).toEqual(16);
+  });
+});
+
+describe("Problem1 test 2", function() {
+	var objectProblem1;
+	var testArray;
+	
+	beforeEach(function() {
+		objectProblem1 = new ProblemsSolved();
+	});
+  
+  it("Sum of even numbers in case of not exist even number/numbers", function() {
+	testArray = [1,3,1,1,5];
+	expect(objectProblem1.SumOfEvenNumberInArray(testArray)).toEqual(0);
+  });
+});
+
+describe("Problem1 test 3", function() {
+	var objectProblem1;
+	var testArray;
+	
+	beforeEach(function() {
+		objectProblem1 = new ProblemsSolved();
+	});
+  
+  it("Sum of even numbers in case of size of array is 0", function() {
+	testArray = [];
+	expect(objectProblem1.SumOfEvenNumberInArray(testArray)).toEqual(0);
+  });
+});
```

Comment:
Do you like the following approach with nesting describes ?

````js

describe('Problem1', function () { 
    var instance, subject;
	
    beforeEach(function() {
        instance = new problem1();
    });

    describe('SumOfEvenNumberInArray', function () {
        var testArray;

        beforeEach(function () {
            testArray = [];
            subject = function () {
                return instance.SumOfEventNumberInArray(testArray);
            };
        });

        describe('when array size is 0', function () {
            beforeEach(function () {
                testArray = [];
            });

           it('returns 0', function () {
              exspect(subject()).toEqual(0);
           });
        });

        describe('when array does not contain even numbers', function () {
            beforeEach(function () {
                testArray = [1, 3, 1, 1, 5];
            });

           it('returns 0', function () {
              exspect(subject()).toEqual(0);
           });
        });

        describe('when array contains even numbers', function () {
            beforeEach(function () {
                testArray = [4,4,4,4,5];
            });

           it('returns sum of event numbers', function () {
              exspect(subject()).toEqual(16);
           });
        });
    });
});

````

Do you find this easier to follow ?